GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java

```diff
@@ -774,13 +771,14 @@ public SurveyInstance getRegistrationSurveyInstance(SurveyedLocale locale,
     public List<SurveyInstance> getMonitoringData(@Nonnull List<SurveyedLocale> surveyedLocales, int numberOfInstances) {
         List<SurveyInstance> result = new ArrayList<>();
 
-        surveyedLocales.forEach(surveyedLocale ->
-                result.add(getRegistrationSurveyInstance(surveyedLocale, surveyedLocale.getCreationSurveyId())));
-
-        surveyedLocales.forEach(surveyedLocale ->
+        for (SurveyedLocale s: surveyedLocales) {
```

Comment:
We're changing the logic and introduce a *corner case*. We should add a test in [SurveyInstanceDAOTest](https://github.com/akvo/akvo-flow/blob/757155a918a44db85ec5f64a844cc6527decfec5/GAE/test/com/gallatinsystems/surveyal/dao/SurveyInstanceDAOTest.java)